pkg/store/promproxy.go

```diff
@@ -156,6 +165,36 @@ func (p *PrometheusProxy) Series(ctx context.Context, r *storepb.SeriesRequest)
 	return res, nil
 }
 
+func extLabelsMatches(extLabels labels.Labels, ms []storepb.LabelMatcher) (bool, []storepb.LabelMatcher, error) {
+	elmap := map[string]string{}
+	for _, l := range extLabels {
+		elmap[l.Name] = l.Value
+	}
```

Comment:
Why do we build this map? We have `labels.Labels` so `extLabels.Get(name) == ""` should be a sufficient check for non-existance.

In general for Prometheus an empty label is defined as equivalent to the label not existing. There's one exception that applies during federation – but that's more of a hack we can ignore in general.

Reply:
sorry, forgot about `Get` method.